clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -355,7 +355,12 @@ scheduler_tmpl_t<RecordType, ReaderType>::stream_t::next_record(RecordType &reco
     uintptr_t marker_value;
     if (scheduler_->record_type_is_marker(record, marker_type, marker_value)) {
         switch (marker_type) {
-        case TRACE_MARKER_TYPE_TIMESTAMP: last_timestamp_ = marker_value; break;
+        case TRACE_MARKER_TYPE_TIMESTAMP:
+            last_timestamp_ = marker_value;
+            if (first_timestamp_ == 0)
+                first_timestamp_ = last_timestamp_;
```

Comment:
This new "first timestamp" is problematic for the output stream here for the parallel analysis model with multiple shards scheduled but tools wanting to keep them separate.  Unlike the top-level headers, which we kind of assume are all the same (with some corner cases we haven't resolved IIRC), or the most-recent records which work in that model, this will do the wrong thing for a mixed sequence of shards on one output stream for one analyzer worker.  We should discuss what to do about this.

Reply:
As you pointed out offline, SCHEDULER_USE_INPUT_ORDINALS already solves this. I fixed get_last_timestamp in the scheduler to use SCHEDULER_USE_INPUT_ORDINALS too.